### ISSUE TITLE
Is RAdam needed when fitting perfectly a small batch e.g. 500 examples?

### ISSUE THREAD

**Issue by brando90** [LiyuanLucasLiu/RAdam#63]

**Comment by LiyuanLucasLiu** [comment]
Hi the batch size is not the key issue here. The batch number per epoch is. The adaptive learning rate is calculated as 1/[g_0^2 + g_1^2], and views each batch as one sample.

**Comment by brando90** [comment]
> Hi the batch size is not the key issue here. The batch number per epoch is. The adaptive learning rate is calculated as 1/[g_0^2 + g_1^2], and views each batch as one sample.

Hi Liyuan,

Thanks for the reply!

I was asking because I had a transformer that had a lot of difficulties fitting just a synthetic data set of sequences (500 examples) and was trying to figure out what was causing the issue (while attempting to reproduce this paper https://arxiv.org/abs/1911.09983).

**Comment by brando90** [comment]
@LiyuanLucasLiu the quick guide section says:

> Further tune hyper-parameters (including the learning rate) for a better performance.

my understanding of this is that further HP tuning is only needed to improve performance but not basic things like fitting a (small) data set like this one.

Is this your experience too? i.e. if I were to plug in RAdam without a scheduler, annealing, warm up, then you would expect the transformer to be able to fit the data set perfectly?
